torch/_inductor/codegen/common.py

```diff
@@ -202,6 +203,7 @@ def __init__(self, sizevars=None):
         self.output_buffers = dict()
         self.inplace_buffers = dict()
         self.sizevars = sizevars or dict()
+        self.precomputed_sizevars = dict()
```

Comment:
I'd suggest putting this as part of `V.graph.sizevars` rather than here in `KernelArgs`.  That way when calling kernels we can just treating it like a regular `self.sizevar`.

The advantage of that is that if the same expression is needed by multiple kernels we will only compute it once on the host side.  I also think the code might be a bit cleaner since the sizevars code is simpler than the triton calling code.  It will also be easier to do these replacements in indexing since sizevars is global.

Reply:
But here it conveniently keeps only values needed for this kernel, so `python_argdef` produces all the needed args

Reply:
You can just add the computed sizevars to `self.sizevars` and it should work.  I think this actually happens automatically if you put them in indexing formulas (via rename indexing).

Reply:
Addressed

Reply:
So we still need this line?  I'd expect the precomputed sizes could just go in as normal sizevar args.